pjmedia/src/pjmedia/rtcp.c

```diff
@@ -991,12 +996,22 @@ PJ_DEF(void) pjmedia_rtcp_build_rtcp(pjmedia_rtcp_session *sess,
     /* Total lost. */
     expected = pj_ntohl(rr->last_seq) - sess->seq_ctrl.base_seq;
 
-    /* This is bug: total lost already calculated on each incoming RTP!
-    if (expected >= sess->received)
-        sess->stat.rx.loss = expected - sess->received;
-    else
+    /* Calculate packet lost */
+    if (expected > sess->received) {
+        unsigned count = expected - sess->received;
+        unsigned period;
+
+        period = count * sess->dec_pkt_size * 1000 / sess->clock_rate;
+        period *= 1000;
+
+        sess->stat.rx.loss = count;
+        TRACE_((sess->name, "%d packet(s) lost", count));
+
+        /* Update loss period stat */
+        pj_math_stat_update(&sess->stat.rx.loss_period, period);
+    } else {
         sess->stat.rx.loss = 0;
-    */
+    }
```

Comment:
What if this calculation is done on each incoming RTP instead of in building RTCP for faster stats update (by moving this block there perhaps), because the RTCP interval is configurable so it can be long, default is not that long though, 5 seconds.

Also, perhaps only update/calculate it when out-of-order flag is not set.

Reply:
Okay, done in the latest commit.

There are a couple of drawbacks though:
- Calculating only when not out of order means that the packet loss stat may not always be accurate.
- There is also complication with updating the loss period. The stats such as avg, max, std, etc will be inaccurate.

Reply:
Re: first drawback should be acceptable, as I think most RTP packets are ordered (e.g: every second there should be at least two properly ordered packets).

Re: second drawback. Currently the `count` is the total loss count since the beginning of the session? If it is, the period should not be counted using `count`. IMO the loss period represents the consecutive loss packets (in time unit) of every packet loss occurance. So the calculation is perhaps: when total loss count is increased, compare the current loss  count vs previous loss count, the difference should be the period?